### PR TITLE
fix some bugs with ROUTER_SLOWLORIS_TIMEOUT

### PR DESCRIPTION
**Pull request by JacobTanenbaum**
validate inputs from ROUTER_SLOWLORIS_TIMEOUT

Take a suggestion from the comments in https://bugzilla.redhat.com/show_bug.cgi?id=1328037 on timeout
http-keep-alive

### CODE REVIEW

**Comment by JacobTanenbaum** [comment]
@openshift/networking

**Comment by knobunc** [comment]
LGTM [test]

File: images/router/haproxy/conf/haproxy-config.template [hunk 1]
```diff
@@ -39,7 +39,13 @@ defaults
   option httplog
   log global
 {{ end }}
-  timeout http-request {{env "ROUTER_SLOWLORIS_TIMEOUT" "10s" }}
+{{ if (matchPattern "[1-9][0-9]*(us|ms|s|m|h|d)?" (env "ROUTER_SLOWLORIS_TIMEOUT" "")) }}
+  timeout http-request {{env "ROUTER_SLOWLORIS_TIMEOUT" "" }}
+  timeout http-keep-alive 30s
+{{ else }}
+  timeout http-request 10s
+  timeout http-keep-alive 30s
```
Comment (ramr) [inline images/router/haproxy/conf/haproxy-config.template]: Move the `http-keep-alive 30s` out of the `if-else` as its the same.

**Comment by ramr** [comment]
Minor comment ... LGTM

**Comment by JacobTanenbaum** [comment]
@ramr @knobunc PTAL

**Comment by knobunc** [comment]
[test]

**Comment by JacobTanenbaum** [comment]
[test]

**Comment by knobunc** [comment]
LGTM [merge]

**Pull request merged**